### PR TITLE
08 patch api articles article

### PR DESCRIPTION
**Pull request by lemonses**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -169,8 +170,90 @@ describe('GET /api/articles/:article_id/comments',()=>{
...
+        })
+    })
+    test('should return a 400 bad request if inc_votes is invalid data type',()=>{
+        return request(app)
+        .patch('/api/articles/1')
+        .send({ inc_votes : 'hello', not_important:'nothing', test: 100})
+        .expect(400).then(({body})=>{
+            expect(body.message).toBe('Bad request')
+        })
+    })
```
Comment (elijaharley) [inline __tests__/server.test.js:242]: I'd consider grouping all of your 200 tests together and error tests below just for readability. I'd also add a test here for an empty body. Otherwise really comprehensive testing here, good work. 👍

File: server/controllers/articles.controller.js [hunk 2]
```diff
@@ -24,4 +24,16 @@ exports.getComments = (req,res,next) => {
...
+exports.patchArticle = (req,res,next) => {
+    const {body} = req
+    const {article_id} = req.params
+    fetchArticle(article_id).then(()=>{
+        return updateArticle(article_id,body)
+    }).then((article)=>{
+        res.status(200).send({article})
+    }).catch((err) => {
+        next(err)
+    })
```
Comment (elijaharley) [inline server/controllers/articles.controller.js:36]: We don't need to explicitly check that an article exists here - we can handle this functionality in the model -  if the article doesn't exist as it won't return anything and we can use a Promise.reject.

**Review by elijaharley — APPROVED** [review]
Really nice work on this endpoint, please see my comments inline and make the necessary updates, then you're good to merge.
